src/vs/workbench/contrib/terminalContrib/chat/browser/terminalChatService.ts

```diff
@@ -69,6 +69,10 @@ export class TerminalChatService extends Disposable implements ITerminalChatServ
 		}
 	}
 
+	getToolSessionTerminalInstances(): readonly ITerminalInstance[] {
+		return Array.from(this._terminalInstancesByToolSessionId.values());
+	}
```

Comment:
Returning the iterator is better here, we can then use that instead of constructing the array on the outside.